### PR TITLE
Add token parameter to sinv capabilities

### PR DESCRIPTION
**Pull request by lab176**
### Summary of this Pull Request (PR)

Add token identification parameter to cos_sinv()
Increases sinv capability from 32 to 64
Enables use of `cos_asm_server_stub_spdid`  interface stubs

### Intent for your PR

Choose one (Mandatory):

- [] This PR is for a code-review and is intended to get feedback, but not to be pulled yet.
- [x] This PR is mature, and ready to be integrated into the repo.

### Reviewers (Mandatory):
(Specify @<github.com username(s)> of the reviewers. Ex: @user1, @user2)
@RobertGiff @gparmer @Others (I hope Others is peachg)

### Code Quality

As part of this pull request, I've considered the following:

[Style](https://github.com/gparmer/composite/raw/ppos/doc/style_guide/composite_coding_style.pdf):

- [x ] Comments adhere to the Style Guide (SG)
- [x ] Spacing adhere's to the SG
- [ x] Naming adhere's to the SG
- [ x] All other aspects of the SG are adhered to, or exceptions are justified in this pull request
- [ ] I have run the auto formatter on my code before submitting this PR (see doc/auto_formatter.md for instructions)

[Code Craftsmanship](http://www2.seas.gwu.edu/~gparmer/posts/2016-03-07-code-craftsmanship.html):

- [x] I've made an attempt to remove all redundant code
- [x] I've considered ways in which my changes might impact existing code, and cleaned it up
- [x] I've formatted the code in an effort to make it easier to read (proper error handling, function use, etc...)
- [x] I've commented appropriately where code is tricky
- [x] I agree that there is no "throw-away" code, and that code in this PR is of high quality

### Testing

I've tested the code using the following test programs (provide list here):

- tested on iotgw branch's interface component's not included in this PR

### CODE REVIEW

File: src/kernel/include/shared/cos_types.h [hunk 1]
```diff
@@ -255,7 +255,7 @@ enum
                                                          CAPMAX_ENTRY_SZ),
 	BOOT_CAPTBL_SELF_INITHW_BASE   = round_up_to_pow2(BOOT_CAPTBL_SELF_INITRCV_BASE + NUM_CPU_COS * CAP64B_IDSZ,
                                                         CAPMAX_ENTRY_SZ),
-	BOOT_CAPTBL_SINV_CAP           = BOOT_CAPTBL_SELF_INITHW_BASE + CAP32B_IDSZ,
+	BOOT_CAPTBL_SINV_CAP           = BOOT_CAPTBL_SELF_INITHW_BASE + CAP64B_IDSZ,
```
Comment (phanikishoreg) [inline src/kernel/include/shared/cos_types.h]: You're changing the offset of `SINV_CAP` and not the offset of `LAST_CAP` which you want to be based on the new size of `SINV_CAP`.

Comment (lab176) [reply src/kernel/include/shared/cos_types.h]: oh wow, great point. I'll fix now.

File: src/components/implementation/no_interface/llbooter/boot_deps.h [hunk 2]
```diff
@@ -92,7 +92,7 @@ boot_newcomp_sinv_alloc(spdid_t spdid)
 			user_cap_vaddr = (void *) (new_comp_cap_info[spdid].vaddr_mapped_in_booter + (new_comp_cap_info[spdid].vaddr_user_caps - new_comp_cap_info[spdid].addr_start) + (sizeof(struct usr_inv_cap) * i));
 
 			/* Create sinv capability from client to server */
-			sinv = cos_sinv_alloc(newcomp_compinfo, interface_compinfo->comp_cap, (vaddr_t)new_comp_cap_info[spdid].ST_user_caps[i].service_entry_inst);
+			sinv = cos_sinv_alloc(newcomp_compinfo, interface_compinfo->comp_cap, (vaddr_t)new_comp_cap_info[spdid].ST_user_caps[i].service_entry_inst, (unsigned long) spdid);
```
Comment (gparmer) [inline src/components/implementation/no_interface/llbooter/boot_deps.h]: Can we make this not `spdid`?  I'd like to purge this with fire.

`token`, perhaps?

File: src/components/implementation/no_interface/llbooter/boot_deps.h [hunk 3]
```diff
@@ -118,7 +118,7 @@ boot_newcomp_create(spdid_t spdid, struct cos_compinfo *comp_info)
 	new_comp_cap_info[spdid].compinfo->comp_cap = cc;
 
 	/* Create sinv capability from Userspace to Booter components */
-	sinv = cos_sinv_alloc(&boot_info, boot_info.comp_cap, (vaddr_t)__inv_test_entry);
+	sinv = cos_sinv_alloc(&boot_info, boot_info.comp_cap, (vaddr_t)__inv_test_entry, (unsigned long)spdid );
```
Comment (gparmer) [inline src/components/implementation/no_interface/llbooter/boot_deps.h]: ditto.

File: src/components/include/cos_kernel_api.h [hunk 4]
```diff
@@ -113,7 +113,7 @@ typedef void (*cos_thd_fn_t)(void *);
 thdcap_t cos_thd_alloc(struct cos_compinfo *ci, compcap_t comp, cos_thd_fn_t fn, void *data);
 /* Create the initial (cos_init) thread */
 thdcap_t  cos_initthd_alloc(struct cos_compinfo *ci, compcap_t comp);
-sinvcap_t cos_sinv_alloc(struct cos_compinfo *srcci, compcap_t dstcomp, vaddr_t entry);
+sinvcap_t cos_sinv_alloc(struct cos_compinfo *srcci, compcap_t dstcomp, vaddr_t entry, unsigned long token);
```
Comment (gparmer) [inline src/components/include/cos_kernel_api.h]: woot.

File: src/kernel/include/inv.h [hunk 5]
```diff
@@ -20,6 +20,7 @@ struct cap_sinv {
 	struct cap_header h;
 	struct comp_info  comp_info;
 	vaddr_t           entry_addr;
+	unsigned long 	  token;
```
Comment (gparmer) [inline src/kernel/include/inv.h]: Sigh...so annoyed this takes it to the 64 byte size.

File: src/components/implementation/no_interface/llbooter/boot_deps.h [hunk 6]
```diff
@@ -80,6 +80,8 @@ boot_newcomp_sinv_alloc(spdid_t spdid)
 	void *user_cap_vaddr;
 	struct cos_compinfo *interface_compinfo;
 	struct cos_compinfo *newcomp_compinfo = new_comp_cap_info[spdid].compinfo;
+	/* TODO: Purge rest of booter of spdid convention */
+	unsigned long token = (unsigned long)spdid;
```
Comment (lab176) [inline src/components/implementation/no_interface/llbooter/boot_deps.h]: I did this based on your comment on previous commit. The llbooter is infested with spdid, so I figured this felt at least more consistent with current code while acknowledging our intention to switch to token. As opposed to changing the spdid argument for the function all together.

Comment (gparmer) [reply src/components/implementation/no_interface/llbooter/boot_deps.h]: OK fine.  I suppose the fires of `spdid` purgatory will remain unsatiated for now.

**Pull request merged**